optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/scope/PartitionChangeMove.java

```diff
@@ -98,9 +98,7 @@ public boolean isMoveDoable(ScoreDirector<Solution_> scoreDirector) {
 
     @Override
     protected PartitionChangeMove<Solution_> createUndoMove(ScoreDirector<Solution_> scoreDirector) {
-        // HACK Creating an undoMove is a waste of time because it is never used,
-        // but this method is called anyway so it returns null to fail fast if it is used
-        return null;
+        throw new UnsupportedOperationException("Programming error: undo move should not have been called.");
```

Comment:
(nitpick) we typically start such errors with "Impossible state:" to indicate that it's defenitly an optaplanner bug.
```suggestion
        throw new UnsupportedOperationException("Impossible state: undo move should not be called.");
```